dht/range_streamer.cc

```diff
@@ -266,8 +266,7 @@ future<> range_streamer::stream_async() {
                 unsigned nr_ranges_streamed = 0;
                 size_t nr_ranges_total = range_vec.size();
                 size_t nr_ranges_per_stream_plan = nr_ranges_total / 10;
-                dht::token_range_vector ranges_to_stream;
-                auto do_streaming = [&] {
+                auto do_streaming = [&] (dht::token_range_vector&& ranges_to_stream) {
```

Comment:
Would be even simpler to reason about if you used a value parameter instead of reference. The code wouldn't have to rely on temporary object lifetime extension.

Reply:
Fair point.
I see you already queued the PR.
If needed, I'll fix it as a follow up

Reply:
I think this is fine as well.